include/circt/Dialect/FIRRTL/FIRRTLStructure.td

```diff
@@ -248,4 +248,51 @@ def FMemModuleOp : FIRRTLModuleLike<"memmodule"> {
   let hasCustomAssemblyFormat = 1;
 }
 
+def FInterfaceOp : FIRRTLModuleLike<"interface", [SingleBlock, NoTerminator]> {
+  let summary = "FIRRTL Interface";
+  let description = [{
+    The "firrtl.interface" operation represents an SV Interface. It is exactly
+    like an FModuleOp. LowerToHW lowers this to an sv.interface.
```

Comment:
Is it really "exactly like"?  Would this be equivalent to putting an attribute on an FModuleOp indicating it's really an interface (other than of course the way code interprets the contents)? Not saying that's what should be done, just asking, haven't poked around in an editor yet /looked closely through this.

From short test including examples, looks like this primarily is meant to hold declarations of signals.

Consider what happens if symbols/annotations/etc target this-- I imagine we don't want to, say, inline/flatten this.

Reply:
True, its not "exactly like". We need to verifier to make sure, that only signal declarations, verbatims, maybe subfields, and maybe some RefType ops will be allowed here. 

`firrtl.interface` as an annotation target is a very interesting and maybe important consideration here. (not handled in this PR yet). Because, right now the interfaces don't exist in the input `fir` , hence no way for chisel to target them. But existing designs do use the "generated interface" as target, and as a workaround, `internal_path` (opaque string paths) are used to target them, the path to signals inside the interface are hard coded in the design !!